fastlane/Fastfile

```diff
@@ -1159,6 +1164,45 @@ platform :ios do
     )
   end
 
+  private_lane :suppress_deprecation_warnings_in_headers do |options|
+    xcframework_path = options[:xcframework_path]
+    
+    # Find all RevenueCat-Swift.h files in the XCFramework
+    swift_headers = Dir.glob("#{xcframework_path}/**/RevenueCat-Swift.h")
+    
+    swift_headers.each do |header_path|
+      UI.message("Processing header: #{header_path}")
+      
+      # Read the current header content
+      content = File.read(header_path)
+      
+      # Check if pragma directives are already present
+      unless content.include?('#pragma clang diagnostic ignored "-Wdeprecated-declarations"')
+        # Add pragma directives to suppress StoreKit 1 deprecation warnings
+        # Insert after the existing pragma clang diagnostic push
+        pragma_insertion_point = content.index('#pragma clang diagnostic ignored "-Wgcc-compat"')
+        
+        if pragma_insertion_point
+          insertion_index = content.index("\n", pragma_insertion_point) + 1
+          pragma_directives = <<~PRAGMA
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+          PRAGMA
+          
+          # Insert the pragma directives
+          content.insert(insertion_index, pragma_directives)
+          
+          # Write back to file
+          File.write(header_path, content)
+          UI.success("Added deprecation warning suppression to #{File.basename(header_path)}")
+        else
+          UI.warning("Could not find insertion point in #{File.basename(header_path)}")
```

Comment:
I think this should fail, otherwise we will most likely miss this

Reply:
Great point. I originally didn't consider it an error because not being able to insert the suppression of deprecation warnings is not a huge problem. But I agree that if we are doing it, we should be aware if it breaks in the future. And a warning would be very easy (and likely) to miss. Thanks for the improvement!